### PR TITLE
Add model for Seasonal Global Trend

### PR DESCRIPTION
**Pull request by neerajprad**
This seeds the repo with some time series models that @fritzo and I are working on. This is just based on my progress up till this point. I will continue to work on it:

 - Work with other / longer time series datasets.
 - Use @martinjankowiak's SVGD algorithm.
 - Better uncertainty estimation / projection.
 - Currently, the last model takes too long to compile for HMC. We may need to use NumPyro if we want to compare MCMC on more models / datasets.

NB can be viewed at https://nbviewer.jupyter.org/github/pyro-ppl/sandbox/blob/45e2eb308459c11c9ecbf14aa23a9fbc0783197c/2019-08-time-series/time_series_sgt.ipynb

### CODE REVIEW

**Comment by neerajprad** [comment]
@martinjankowiak - I have tried to use SVGD on this problem, and I found the following issues:

 - the algorithm is quite sensitive to Adam's learning rate. If it is too high the gradient norms seem to  fluctuate quite a bit, and if it is low then convergence seems to be slow. Are there other optimizers that you would recommend trying? Or maybe I just need to use a really small learning rate and run for many more iterations?
 - on an earlier version of some of these models, SVGD was arguably working with the mean square gradients converging near 0, but I was still getting a somewhat flat trend-line, and the result was quite different from HMC (just imagine the HMC curves with seasonality trends squished by 70%). Would you expect that?

**Comment by fehiepsi** [comment]
@neerajprad I can recall that the HMC result is similar to an early version of numpyro notebook. I think that [generalized seasonality and seasonal average method in model options](https://cran.r-project.org/web/packages/Rlgt/vignettes/GT_models.html) will help a lot. Original SGT model gives RMSE 1500 while incorporates these techniques reduce RMSE to 1250, which is quite decent comparing to other prediction tools. However, I can see that `model_sgt_2` gives the mean plot similar to the current version in numpyro, so it would be nice to see its RMSE. As this is a workshop tutorial, I think it is better to put concrete names for variables to help audience interpret the model. The models and prediction look reasonable to me. Though for prediction, it might be better to clamp the output by 0 because those models suppose `y` is positive. In addition, using `median` (instead of `mean`) of predictions might be helpful for this model (especially for high variance predictions). :)

**Comment by martinjankowiak** [comment]
@neerajprad it's hard to say. i myself haven't much empirical experience with SVGD and have run into some strange behavior. some things you might try:

- use a smallish number of particles (e.g. 10; makes the optimization easier?)
- decay the learning rate over the course of optimization (start high, end low)
- [modulate the kernel bandwidth over the course of optimization](https://github.com/pyro-ppl/pyro/blob/97e60edc80527851b2a5c4363c9c3219de87a5f7/tests/infer/test_svgd.py#L38)

**Comment by fritzo** [comment]
@neerajprad Could we rename this notebook to part_ii_inference.ipynb? I think it would clarify our three tutorials:
- part_i_models.ipynb
- part_ii_inference.ipynb
- part_iii_custom.ipynb

**Comment by neerajprad** [comment]
@fehiepsi, @martinjankowiak - Thanks for your feedback! I am now looking at the unemployment claims time series data-set which is a bit longer in #6, and will incorporate your comments / suggestions in that.
